### PR TITLE
feature: support the READ_COMMITED isolation for GlobalLock

### LINKED ISSUE #3068: Trigger check global lock automatically when use @GlobalLock
## Why you need it?
For now when use @GlobalLock, have to add "for update" in the sql. I hope we can add “for update” automatically when use @GlobalLock with select.

If you want to optimize it, please comment "Please assign it to me" is this issue.
## Other related information

### PR DESCRIPTION
**Pull request by caohdgege**
### Ⅰ. Describe what this PR did
add the isolation for GlobalLock annotation, now support READ_UNCOMMITTED & READ_UNCOMMITED。

为GlobalLock注解添加isolation的参数，目前支持了READ_UNCOMMITTED 和 READ_UNCOMMITED两种模式。

### Ⅱ. Does this pull request fix one issue?

fixes #3068

### CODE REVIEW

**Review by a364176773 — COMMENTED** [review]
在globaltrasactional注解中也加上隔离级别？

File: core/src/main/java/io/seata/core/model/GlobalLockConfig.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
     private int lockRetryTimes;
 
+    private boolean autoAddForUpdate;
```
[the remark this answers is not in the document]
Comment (caohdgege) [reply core/src/main/java/io/seata/core/model/GlobalLockConfig.java]: OK

File: spring/src/main/java/io/seata/spring/annotation/GlobalLock.java [hunk 2]
```diff
@@ -53,4 +53,10 @@
      * @return
      */
     int lockRetryTimes() default -1;
+
+    /**
+     * if autoAddForUpdate is true
+     *  then it will check lock even the target sql do not contain for update
+     */
+    boolean autoAddForUpdate() default false;
```
[the remark this answers is not in the document]
Comment (caohdgege) [reply spring/src/main/java/io/seata/spring/annotation/GlobalLock.java]: default READ_UNCOMMITED

File: core/src/main/java/io/seata/core/constants/Isolation.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    READ_COMMITTED,
+    /**
+     * the isolation repeatable read
+     */
+    REPEATABLE_READ,
+    /**
+     * the isolation serializable
+     */
+    SERIALIZABLE,
```
Comment (a364176773) [inline core/src/main/java/io/seata/core/constants/Isolation.java:37]: SERIALIZABLE;

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Pull request closed without merging**